### PR TITLE
DAS-2214: Update environment to exclude conda's default channel.

### PR DESCRIPTION
**Pull request by flamingbear**
This ensures we don't run into licensing issues.

## Description

Simply removes defaults and replaces with nodefaults so that all packages are pulled from conda-forge rather than the pay-to-use default channel.

## Jira Issue ID

DAS-NONE

## Local Test Steps

pull and build all images and ensure the regression tests all pass in the UAT environment. 

You will need to set the environment:

```
export HARMONY_HOST_URL=https://harmony.uat.earthdata.nasa.gov
export AWS_ACCESS_KEY_ID=[yours]
export AWS_SECRET_ACCESS_KEY=
export EDL_USER=[yours]
export EDL_PASSWORD=[yours]
```

```
make images
./run-notebooks.sh
```

## PR Acceptance Checklist
* [N/A] Acceptance criteria met
* [N/A] Tests added/updated (if needed) and passing
* [N/A] Documentation updated (if needed)
* [X] CHANGELOG updated with the changes for this PR

### CODE REVIEW

**Comment by flamingbear** [comment]
@owenlittlejohns I think this is a good change, but does it need new versions of all the images, or can they just get rolled up as they are changed in the future?

This is simply ensuring we don't use the conda defaults channel for any of our tests.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ versioning. Rather than a static releases, this repository contains of a number
 of regression tests that are each semi-independent.  This CHANGELOG file should be used
 to document pull requests to this repository.
 
+## 2024-11-04 ([#107](https://github.com/nasa/harmony-regression-tests/pull/112))
```
Comment (owenlittlejohns) [inline CHANGELOG.md:8]: Super nit: This should have `[#112]`, I think.

Comment (flamingbear) [reply CHANGELOG.md]: it should [1d79e97](https://github.com/nasa/harmony-regression-tests/pull/112/commits/1d79e970f9dfcdf86e13be5b801da64524ddd180)

**Review by owenlittlejohns — APPROVED** [review]
Looks good to me. I managed to build all the images and run them locally. This is definitely a good change to make with licensing concerns surrounding conda packages from the defaults channel!

Also, to capture some comments in Slack regarding whether to publish new images for each test suite:

* Generally, I don't feel super strongly, but would lean towards publishing all the affected images.
* It's probably good to isolate these changes from other future changes, in case something breaks in the next published image for a regression test service.
* Some test suites don't get updated frequently, so they may not use these changes if we don't publish now.
* The CI/CD that is automatically triggered when new service versions are released via the `/service-image-tag` endpoint uses the Docker images tagged with `latest`. So publishing now will get those new images used immediately.

**Pull request merged**